### ISSUE TITLE
[FR] Support displaying an image instead of icon

### ISSUE THREAD

**Issue by ildar170975** [elchininet/keep-texts-in-tabs#110]
Use-case: each view represents some user, that is why I need to show an image instead of icon in a toolbar. 

It is possible with card-mod-theme:
https://community.home-assistant.io/t/card-mod-super-charge-your-themes/212176/1112?u=ildar_gabdullin

I can show either only an image or with a text:
![](https://community-assets.home-assistant.io/original/4X/8/2/0/8205da513e90b758f00db36918a9f9d7ba225ed2.png)

![](https://community-assets.home-assistant.io/original/4X/6/9/d/69d842b9c312e6e9ecbdb048e4a781f1620c4e0b.png)

Is it possible to add a corresponding functionality to this plugin?
Assume a user adds a new option to a view - “image” (contains some path), and then plugin uses this image instead of an icon.

**Comment by elchininet** [comment]
Hi @ildar170975,

The intention behind this plugin is to do what its name says and no more 🙂

But what you are proposing sounds interesting for something that I have been planning to do for a while, reviving [custom-header](https://github.com/maykar/custom-header) (I talked with @Mariusthvdb about it in [this thread](https://github.com/NemesisRE/kiosk-mode/discussions/119#discussioncomment-7185891)).

But I would not fork `custom-header`, it should be a plugin aimed to do personalisations only in the header and without bloating it with specific user cases to avoid it became what Maykar described in [its farewell](https://community.home-assistant.io/t/custom-header/155399/1100).

I'll let you know, I think that custom header is something that many users miss.

**Comment by ildar170975** [comment]
@elchininet 
Thanks for a feedback)
Never used a custom header; my 1st impression was like “what a huge plugin, I do not need it”, and you have just confirmed my fears by this post of Mayker.
I myself use card-mod-themes a little and prefer to use external js-files - less transitions.
Just wondered if that “image” feature can be implemented… Now thinking about my own PR to HA Frontend to support images.

**Comment by ildar170975** [comment]
Feel free to close this issue!

**Comment by elchininet** [comment]
>Feel free to close this issue!

I'll let it open, it will be a reminder to not forget about this topic 🙂

**Comment by Mariusthvdb** [comment]
custom-header had numbered views in a dashboard....
it so happens today I needed a way to count the views in a dashboard. would that be somehow possible currently
(sorry for off topic...)

**Comment by elchininet** [comment]
>custom-header had numbered views in a dashboard....

What do you mean with "numbered views in a dashboard"?

**Comment by Mariusthvdb** [comment]
we could select the view/tab by their number.
set icon x for tab 3, icon z for tab 4 etc etc. not sure about the exact syntax anymore.

I remembered that, and it made we want to count the number of views in a dashboard, and I had hoped there would be some javascript trick template available

**Comment by elchininet** [comment]
>we could select the view/tab by their number.

Yes, that was a config set settings taking into account the view index, but this didn't count the sidebar items.

>I remembered that, and it made we want to count the number of views in a dashboard, and I had hoped there would be some javascript trick template available

I can give you a script to paste in your dev console to count the views, or are you talking about using the number in templates?

**Comment by Mariusthvdb** [comment]
testing you custom-sidebar, I tried to create a notification badge on the menu items (dashboards), with the count of their views. So yes, that would probably require a template in the config.

**Comment by elchininet** [comment]
>testing you custom-sidebar, I tried to create a notification badge on the menu items (dashboards), with the count of their views. So yes, that would probably require a template in the config.

Then it should be a `JavaScript` template because it is a front-end code (the server doesn't know anything about what do you have on your DOM). And of course, it will not be updated when you change from one dashboard to another. But apart from that yes you can do it querying your DOM.
